### PR TITLE
Improve batch swap transfer

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/liquidity-proxy/src/lib.rs [hunk 1]
```diff
@@ -1964,24 +1973,32 @@ impl<T: Config> Pallet<T> {
                     fail!(Error::<T>::InvalidReceiversInfo);
                 }
 
-                let out_amount = receivers.iter().map(|recv| recv.target_amount).sum();
+                let out_amount = receivers
+                    .iter()
+                    .map(|recv| recv.target_amount)
+                    .sum::<Balance>()
```
Comment (wer1st) [inline pallets/liquidity-proxy/src/lib.rs:1979]: `sum()` can overflow and panic

File: rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "nightly-2023-03-21"
+targets = [ "wasm32-unknown-unknown" ]
+profile = "default"
```
Comment (wer1st) [inline rust-toolchain.toml:4]: add empty line please

File: rust-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
```
Comment (Alexey-N-Chernyshov) [inline rust-toolchain.toml:1]: As I remember, there was a reason not to commit toolchain file. I could be wrong.

**Review by wer1st — APPROVED** [review]

File: rust-toolchain.toml [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+channel = "nightly-2023-03-21"
```
Comment (Alexey-N-Chernyshov) [inline rust-toolchain.toml:2]: When we decide to change toolchain, we must update this line and [this line](https://github.com/sora-xor/sora2-network/blob/master/housekeeping/docker/env/Dockerfile#L47).
Probably we want to do it now for #709

File: rust-toolchain.toml (hunk 2 above)
Comment (Alexey-N-Chernyshov) [inline rust-toolchain.toml:4]:
```suggestion
profile = "default"

```

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: rust-toolchain.toml (hunk 3 above)
Comment (vovac12) [reply rust-toolchain.toml:1]: Actually it doesn't build without it. And I think we can use it to ensure that correct Rust version is used by everyone. If we will have any problems with this solution we can remove this file

File: rust-toolchain.toml (hunk 4 above)
Comment (vovac12) [reply rust-toolchain.toml:2]: We have the same version of Rust in docker and we can't update it before Substrate update because of problems with wasm

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**